exotica/src/Problems/SamplingProblem.cpp

```diff
@@ -124,7 +124,14 @@ bool SamplingProblem::isValid(Eigen::VectorXdRefConst x)
     scene_->Update(x);
     for (int i = 0; i < NumTasks; i++)
     {
-        Tasks[i]->update(x, Phi.data.segment(Tasks[i]->Start, Tasks[i]->Length));
+        if (Rho(i) != 0)
+        {
+            Tasks[i]->update(x, Phi.data.segment(Tasks[i]->Start, Tasks[i]->Length));
+        }
+        else
+        {
+            Phi.data.segment(Tasks[i]->Start, Tasks[i]->Length).setZero();
```

Comment:
This line will cause line 136 below to fail when the ```TaskSpaceVector``` Phi contains any quaternion rotations.
Setting the data to zero will set ```xyzw``` to zero creating an invalid rotation.
There is no easy way to directly set the sub segment to zero rotation, except:
Somewhere in the initialization, after the Phi was constructed, create a copy of it and set the copy to zero:
```
TaskSpaceVector ZeroTask = Phi;
ZeroTask.setZero()
```
This will set all the rotations to their correct zero values.
Then in the line above do:
```
Phi.data.segment(Tasks[i]->Start, Tasks[i]->Length) = ZeroTask.data.segment(Tasks[i]->Start, Tasks[i]->Length);
```
This appears in every problem type below as well.

The rest of the PR looks good.

Reply:
Actually, an even cheaper option is to completely skip this line. It will get multiplied by a zero so any valid value can stay in Phi.

Reply:
Good point, thank you. I will update accordingly.